pkgs/tools/security/keybase/default.nix

```diff
@@ -1,22 +1,20 @@
-{ stdenv, lib, buildGoPackage, fetchFromGitHub, cf-private
+{ stdenv, lib, buildGoPackage, fetchurl, cf-private
 , AVFoundation, AudioToolbox, ImageIO, CoreMedia
 , Foundation, CoreGraphics, MediaToolbox
 }:
 
 buildGoPackage rec {
   name = "keybase-${version}";
-  version = "2.10.1";
+  version = "3.0.0";
 
   goPackagePath = "github.com/keybase/client";
   subPackages = [ "go/keybase" ];
 
   dontRenameImports = true;
 
-  src = fetchFromGitHub {
-    owner  = "keybase";
-    repo   = "client";
-    rev    = "v${version}";
-    sha256 = "1gfxnqzs8msxmykg1zrhrrl2slmb29gl7b8s4m2g44zxaj91gfi9";
+  src = fetchurl {
```

Comment:
why this change?

Reply:
Using `fetchFromGitHub` does not work on Darwin. Please see 8da3c3648a0279f78a130abe10941a68fec7240a for context.

Reply:
Why not just redefine` fetchFromGitHub` as(simplified):
```nix
{
   fetchFromGitHub = {owner, repo, rev, sha256}: fetchurl {
      inherit sha256;
      url = "https://github.com/${owner}/${repo}/archive/${rev}.tar.gz"
   };
}
```
Then?
There are plenty of packages using the function(which AFAIK was considered the best practice), and so we'd need to change all invocations in `nixpkgs` to ensure consistency and compatibility with MacOS.

Reply:
We'd get a major loss there as the automatically generated archives are not stable and can result in a different hash, which is something we don't want. It's less a problem to just use fetchurl for darwin and open that risk as this would never build on that platform if we didn't.

Commit msg for context
> fetchFromGitHub and thus fetchzip hashes the contents of the archive and
not the archive itself. Unicode file names lead to different checksums
on HFS+ vs. other file systems because of Unicode normalisation

Though it might be worth checking if we can eliminate this problem in the near future by identifying the problematic file upstream.

Reply:
> Though it might be worth checking if we can eliminate this problem in the near future by identifying the problematic file upstream.

I did not like this solution either, I would love to find a way for us to hash the contents regardless of the Filesystem. Keep in mind also that most Darwin machines are now sold pre-installed with APFS, which does not suffer from this issue.